clash-lib/src/Clash/Core/Subst.hs

```diff
@@ -622,6 +623,18 @@ deShadowAlt ::
   (Pat, Term)
 deShadowAlt is = substAlt "deShadowAlt" (mkSubst is)
 
+-- | Ensure that non of the binders of a let-expression shadow each other, nor
```

Comment:
> Ensure that non of the binders of a let-expression shadow each other

Probably true, but `deshadowLetExpr` shouldn't document this behavior as let-bindings with duplicate binders aren't well-defined anyway.